de.metas.payment.esr/src/main/java/de/metas/payment/esr/dataimporter/impl/camt54/ESRDataImporterCamt54.java

```diff
@@ -124,294 +100,111 @@ public ESRDataImporterCamt54(@NonNull final I_ESR_Import header, @NonNull final
 		{
-			if (isReversal(ntry))
-			{
-				trxBuilder.trxType(ESRConstants.ESRTRXTYPE_ReverseBooking);
-			}
-			else
+			while (reader.hasNext())
 			{
-				trxBuilder.trxType(ESRConstants.ESRTRXTYPE_CreditMemo);
+				int event = reader.next();
+				if (XMLStreamConstants.START_ELEMENT == event && reader.getNamespaceCount() > 0)
+				{
+					for (int nsIndex = 0; nsIndex < reader.getNamespaceCount(); nsIndex++)
+					{
+						final String nsId = reader.getNamespaceURI(nsIndex);
+						if (nsId.startsWith("urn:iso:std:iso:20022:tech:xsd"))
+						{
+							return nsId;
+						}
+					}
+				}
 			}
 		}
-		else
-		{
-			// we get charged; currently not supported
-			final IMsgBL msgBL = Services.get(IMsgBL.class);
-			trxBuilder.trxType(ESRConstants.ESRTRXTYPE_UNKNOWN)
-					.errorMsg(msgBL.getMsg(Env.getCtx(), MSG_UNSUPPORTED_CREDIT_DEBIT_CODE_1P, new Object[] { ntry.getCdtDbtInd() }));
-		}
-	}
-
-	private BigDecimal getCrdDbtMultiplier(@NonNull final CreditDebitCode creditDebitCode)
-	{
-		if (creditDebitCode == CreditDebitCode.CRDT)
-		{
-			return BigDecimal.ONE;
-		}
-		return BigDecimal.ONE.negate();
-	}
-
-	private BigDecimal getRvslMultiplier(@NonNull final ReportEntry8 ntry)
-	{
-		if (isReversal(ntry))
-		{
-			return BigDecimal.ONE.negate();
-		}
-		return BigDecimal.ONE;
-	}
-
-	private boolean isReversal(@NonNull final ReportEntry8 ntry)
-	{
-		return ntry.isRvslInd() != null && ntry.isRvslInd();
-	}
-
-	/**
-	 * Verifies that the currency is consistent.
-	 * 
-	 * @param txDtls
-	 * @param trxBuilder
-	 */
-	private void verifyTransactionCurrency(
-			@NonNull final EntryTransaction8 txDtls,
-			@NonNull final ESRTransactionBuilder trxBuilder)
-	{
-		if (header.getC_BP_BankAccount_ID() <= 0)
-		{
-			return; // nothing to do
-		}
-
-		// TODO: this does not really belong into the loader! move it to the matcher code.
-		final ActiveOrHistoricCurrencyAndAmount transactionDetailAmt = txDtls.getAmt();
-
-		final String headerCurrencyISO = header.getC_BP_BankAccount().getC_Currency().getISO_Code();
-		if (!headerCurrencyISO.equalsIgnoreCase(transactionDetailAmt.getCcy()))
+		catch (XMLStreamException e)
 		{
-			final IMsgBL msgBL = Services.get(IMsgBL.class);
-			trxBuilder.errorMsg(msgBL.getMsg(Env.getCtx(), MSG_BANK_ACCOUNT_MISMATCH_2P,
-					new Object[] { headerCurrencyISO, transactionDetailAmt.getCcy() }));
+			throw AdempiereException.wrapIfNeeded(e);
 		}
+//		finally 
+//		{
+//            if (reader != null) 
+//            {
+//                try
+//				{
+//					reader.close();
+//				}
+//				catch (XMLStreamException e)
+//				{
+//					throw AdempiereException.wrapIfNeeded(e);
+//				}
+//            }
+//		}
+		
+		return null;
 	}
-
-	private BankToCustomerDebitCreditNotificationV06 loadXML()
+	
+	@Override
+	public ESRStatement importData()
 	{
-		final Document document;
 		try
 		{
+		
 			final XMLInputFactory xif = XMLInputFactory.newInstance();
 			final XMLStreamReader xsr = xif.createXMLStreamReader(input);
```

Comment:
i don't see where you make sure that this reader is closed. Did I miss it?
Why don't you use a try-with-ressources block?

Reply:
How can the reader be open already, since now I am creating it?

Reply:
i'm not concerened about whether the reader is already openend, but rather about whether its closed..or what am I missing?